bgpd/bgp_clist.c

```diff
@@ -986,6 +1025,58 @@ struct lcommunity *lcommunity_list_match_delete(struct lcommunity *lcom,
 	return lcom;
 }
 
+/* Delete all permitted large communities in the list from com.  */
+struct ecommunity *ecommunity_list_match_delete(struct ecommunity *ecom,
+						struct community_list *list)
+{
+	struct community_entry *entry;
+	uint32_t com_index_to_delete[ecom->size];
+	uint8_t *ptr;
+	int delete_index = 0;
+	int i;
+	struct ecommunity local_ecom = {0};
+	struct ecommunity_val local_eval = {0};
+
+	local_ecom.size = 1;
+	/* Loop over each ecommunity value and evaluate each against the
+	 * community-list.  If we need to delete a community value add its index
+	 * to com_index_to_delete.
+	 */
+	for (i = 0; i < (int)ecom->size; i++) {
```

Comment:
Why not `uint32_t i` instead of `int`?

Reply:
There are really that many communities to delete? I wonder why?

Reply:
@ton31337, indeed it should be `uint_32_t i` , fixed.

@riw777, when you have such configuration:
`bgp extcommunity-list standard r1 permit soo 2.2.2.2:2 soo 4.4.4.4:4 nt 4.4.4.4:0 rt 4.4.4.4:4`
you end up with multiple ecommunities to delete.